lia-ui/src/main/java/com/lithium/community/android/ui/components/adapters/LiConversationAdapter.java

```diff
@@ -169,136 +185,317 @@ private void showPopupMenu(final View view, final int position) {
             popup.getMenu().removeItem(R.id.li_action_delete);
         }
 
-        popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
-            @Override
-            public boolean onMenuItemClick(MenuItem menuItem) {
-                if (menuItem.getItemId() == R.id.li_action_report_abuse) {
-                    Bundle bundle = new Bundle();
-                    bundle.putString(LiSDKConstants.REPORT_ABUSE_MESSAGE_ID, item.getId() + "");
-                    bundle.putString(LiSDKConstants.REPORT_ABUSE_AUTHOR_ID, item.getAuthor().getId() + "");
-                    Intent i = new Intent(activity, LiReportAbuseActivity.class);
-                    i.putExtras(bundle);
-                    activity.startActivity(i);
-                    return true;
-                } else if (menuItem.getItemId() == R.id.li_action_mark_read) {
-                    if (item.getUserContext() != null) {
-                        boolean isMarkUnread = false;
-                        if (item.getUserContext() != null && item.getUserContext().getRead()) {
-                            isMarkUnread = true;
+        boolean showSubscribe = (position == 0);
+        boolean isSubscribed = item.getUserContext().isSubscribed();
+        popup.getMenu().findItem(R.id.li_action_subscribe).setVisible(showSubscribe);
+        popup.getMenu().findItem(R.id.li_action_subscribe).setTitle(isSubscribed ? R.string.li_unsubscribe : R.string.li_subscribe);
+        popup.setOnMenuItemClickListener(new PopupMenuClickListener(item, getItems().size() == 1 || position == 0, position));
```

Comment:
would prefer.. boolean isParentItem = getItems().size() == 1 || position == 0;
for better readability

Reply:
👍 sure Sir